crates/biome_js_analyze/src/lint/nursery/no_useless_length_check.rs

```diff
@@ -0,0 +1,366 @@
+
+/// Extract the expressions that perform length comparisons corresponding to the errors you want to check.
+fn get_comparing_length_exp(
+    binary_exp: &JsBinaryExpression,
+    expect_error: &ErrorType,
+) -> Option<AnyJsExpression> {
+    let left = binary_exp.left().ok()?;
+    let operator = binary_exp.operator().ok()?;
+    let right = binary_exp.right().ok()?;
+
+    // Check only when the number appears on the right side according to the original rules.
+    // We assume that you have already complied with useExplicitLengthCheck
+    let compare_exp = left;
+    let value_exp = right;
+
+    let AnyJsExpression::JsStaticMemberExpression(member_exp) = compare_exp else {
+        return None;
+    };
+    let target = member_exp.object().ok()?;
+    let member = member_exp.member().ok()?;
+    if member.text() != "length" || member_exp.is_optional_chain() {
+        return None;
+    }
+    let AnyJsExpression::AnyJsLiteralExpression(AnyJsLiteralExpression::JsNumberLiteralExpression(
+        literal,
+    )) = value_exp
+    else {
+        return None;
+    };
+    let number = literal.as_number()?.round() as i64;
+    // .length === 0
+    if matches!(expect_error, ErrorType::UselessLengthCheckWithEvery)
+        && literal.syntax().text_trimmed() == "0"
+        && (operator == JsBinaryOperator::StrictEquality || operator == JsBinaryOperator::LessThan)
+    {
+        return Some(target);
+    }
+    // .length !== 0
+    if matches!(expect_error, ErrorType::UselessLengthCheckWithSome)
+        && (literal.syntax().text_trimmed() == "0"
+            && (operator == JsBinaryOperator::StrictInequality
+                || operator == JsBinaryOperator::GreaterThan)
+            || number > 0 && operator == JsBinaryOperator::StrictEquality
+            || literal.syntax().text_trimmed() == "1"
+                && operator == JsBinaryOperator::LessThanOrEqual)
+    {
+        return Some(target);
+    }
+    None
+}
+
+pub type Replacer = (JsLogicalExpression, AnyJsExpression, TextRange);
+
+/// Search for logical expressions and list expressions that compare to 0 and Array APIs (`.some()`, `.every()`).
+fn search_logical_exp(
+    any_exp: &AnyJsExpression,
+    replacer: Option<Replacer>,
+    expect_error: &ErrorType,
+    comparing_zeros: &mut HashMap<String, Vec<Replacer>>,
+    array_tokens_used_api: &mut HashSet<String>,
+) -> Option<()> {
+    match any_exp {
+        // || or &&
+        AnyJsExpression::JsLogicalExpression(logical_exp) => {
+            let operator = match expect_error {
+                ErrorType::UselessLengthCheckWithEvery => T![||],
+                ErrorType::UselessLengthCheckWithSome => T![&&],
+            };
+            if logical_exp.operator_token().ok()?.kind() != operator {
+                return None;
+            };
+            let left = logical_exp.left().ok()?;
+            let left_replacer = (logical_exp.clone(), logical_exp.right().ok()?, left.range());
+            search_logical_exp(
```

Comment:
This is a **recursion**, and when you want to avoid it. Usually, recursions are fine, but they **aren't fit** for working with trees, like an AST/CST. 

You should refactor this code and try to use a simple loop, or maybe you could evaluate an iterator.

Reply:
I want to search through the entire logical expression. However, I couldn't think of a method other than recursion to extract `A`, `B`, `C` from `A || B || C` or `A`, `B`, `C`, `D` from `(A || B) || (C || D)`. If it encounters something other than a logical expression or a different operator, the implementation is designed to terminate the process. Therefore, I believe the search will not become heavier than the logical expression being passed. If there is a good utility for this, I would like to use it.